frame/nfts/src/lib.rs

```diff
@@ -94,7 +112,7 @@ pub mod pallet {
 			+ IsType<<Self as frame_system::Config>::RuntimeEvent>;
 
 		/// Identifier for the collection of item.
-		type CollectionId: Member + Parameter + MaxEncodedLen + Copy;
+		type CollectionId: Member + Parameter + MaxEncodedLen + Copy + Default + Incrementable;
```

Comment:
Why is `Default` necessary here?

Reply:
Because we are accessing the `NextCollectionId` like this: `NextCollectionId::<T, I>::get()`, so it requires the `Default` trait. The default value here is something that the `NextCollectionId` starts to increment from(for integers it starts from `0`).

Reply:
Okay, this isn't great, because there's no sensible `Default` for `MultiLocation`. What you are essentially trying to accomplish here is to ask the `CollectionId` about what to do when `NextCollectionId` is empty.

Instead of reaching out to `Default`, I propose adding another associated function to `Incrementable`, something like `initial_value`, which does what the name suggests, and change the `NextCollectionId` to use `OptionQuery`.